File: wealth-pulse-api/src/main/java/com/litchi/wealth/entity/StockMarketHistory.java
package com.litchi.wealth.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.util.Date;

/**
* @description 股票历史行情表
* @author Embrace
* @git: https://github.com/embarce
* @date 2026-02-05
*/
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("tb_stock_market_history")
@Schema(name="StockMarketHistory对象", description="股票历史行情表")
public class StockMarketHistory {


    @TableId(value = "id", type = IdType.AUTO)
    private String id;

    @TableField("stock_code")
    private String stockCode;

    @Schema(name = "tradeDate", description = "交易日期")
    @TableField("trade_date")
    private Date tradeDate;

    @TableField("open_price")
    private BigDecimal openPrice;

    @TableField("high_price")
    private BigDecimal highPrice;

    @TableField("low_price")
    private BigDecimal lowPrice;

    @TableField("close_price")
    private BigDecimal closePrice;

    @Schema(name = "adjClose", description = "复权收盘价")
    @TableField("adj_close")
    private BigDecimal adjClose;

    @TableField("volume")
    private Long volume;


}

File: wealth-pulse-python/app/models/stock_market_history.py
from sqlalchemy import Column, String, Float, Date, BigInteger
from sqlalchemy.orm import Mapped, mapped_column
from app.db.session import Base


class StockMarketHistory(Base):
    __tablename__ = "tb_stock_market_history"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True, comment="History ID")
    stock_code: Mapped[str] = mapped_column(String(20), nullable=False, comment="股票代码")
    trade_date: Mapped[Date] = mapped_column(Date, nullable=False, comment="交易日期")
    open_price: Mapped[float | None] = mapped_column(Float(20), comment="开盘价")
    high_price: Mapped[float | None] = mapped_column(Float(20), comment="最高价")
    low_price: Mapped[float | None] = mapped_column(Float(20), comment="最低价")
    close_price: Mapped[float | None] = mapped_column(Float(20), comment="收盘价")
    adj_close: Mapped[float | None] = mapped_column(Float(20), comment="复权收盘价")
    volume: Mapped[int | None] = mapped_column(BigInteger, comment="成交量")
